### PR TITLE
[sled agent] RSS no longer provisions datasets on M.2 devices

### PR DESCRIPTION
**Pull request by smklein**
Fixes https://github.com/oxidecomputer/omicron/issues/2406

### CODE REVIEW

File: sled-agent/src/storage_manager.rs [hunk 1]
```diff
@@ -1203,17 +1211,49 @@ impl StorageManager {
...
+        let mut zpools = Vec::with_capacity(pools.len());
+
+        for (name, pool) in pools.iter() {
+            let disk_type = if let Some(disk_path) = &pool.disk_path {
+                if let Some(disk) = disks.get(disk_path) {
+                    disk.variant().into()
+                } else {
+                    // If the zpool claims to be attached to a disk that we
+                    // don't know about, that's an error.
+                    return Err(Error::ZpoolNotFound(
```
Comment (leftwo) [inline sled-agent/src/storage_manager.rs:1239]: If we bail here, what happens?   Will the sled agent be able to continue working?

File: sled-agent/src/rack_setup/plan/service.rs [hunk 2]
```diff
@@ -154,7 +157,7 @@ impl Plan {
             //
             // Once this responsibility shifts to Nexus, we actually only
             // need enough zpools to provision CRDB.
-            if zpools.len() < MINIMUM_ZPOOL_COUNT {
+            if zpools.len() < MINIMUM_U2_ZPOOL_COUNT {
```
[the remark this answers is not in the document]
Comment (smklein) [reply sled-agent/src/rack_setup/plan/service.rs:154]: Good call, removed in 17013b0c91e03a8e1a44cdb807cedb3ec64c2a5f

[the remark this answers is not in the document]
Comment (smklein) [reply sled-agent/src/rack_setup/plan/service.rs:154]: For context, I think this comment came from an era when we explicitly enumerated all u.2s -- this is no longer the case, as they generally show up asynchronously

File: sled-agent/src/storage_manager.rs (hunk 1 above)
Comment (smklein) [reply sled-agent/src/storage_manager.rs:1237]: Yes - this is only an error being returned from a querying function, which is only called by RSS surveying the zpool landscape right now.

I could opt to simply ignore the zpool causing the error, but I'm leaning towards "loud errors, returned as early as possible" while we're still in development.

Comment (leftwo) [reply sled-agent/src/storage_manager.rs:1237]: I think being loud about errors is good.
I just was concerned about the rest of sled agent making forward progress if errors happen.

**Review by leftwo — APPROVED** [review]

**Pull request merged**